bids/layout/models.py

```diff
@@ -317,6 +321,11 @@ def get_df(self, include_timing=True, adjust_onset=False):
                     onsets += md['StartTime']
                 data.insert(0, 'onset', onsets)
 
+        if enforce_dtypes:
+            string_cols = {'subject_id', 'session_id', 'participant_id'}
+            for col in string_cols & set(data.columns):
+                data.loc[:, col] = data.loc[:, col].astype(str)
+
```

Comment:
Wouldn't it be too late if original label was `001`? Test you added below doesn't have that

Reply:
Ugh, good catch. I'll pass a dict to `dtype` in `read_csv`. Thanks!

Reply:
This ended up being a bit annoying—the fix is simple, but I had to disable caching for the time being, because the cached data depend on the value of the `enforce_dtypes` argument. This needs to be memoized, but it's not a high priority as reading these files in is not much of a bottleneck.